[15.0][MIG] l10n_es_intrastat_report

flachica opened the pull request:
Depende de:

* [x]  report_xlsx_helper: [[MIG] report_xlsx_helper: migration to 15.0 reporting-engine#584](https://github.com/OCA/reporting-engine/pull/584)
* [x]  intrastat_base: [[MIG] intrastat_base: migration to 15.0 intrastat-extrastat#157](https://github.com/OCA/intrastat-extrastat/pull/157)
* [x]  product_harmonized: [[MIG] product_harmonized: migration to 15.0 intrastat-extrastat#158](https://github.com/OCA/intrastat-extrastat/pull/158)
* [x]  intrastat_product: [[MIG] intrastat_product: migration to 15.0 intrastat-extrastat#165](https://github.com/OCA/intrastat-extrastat/pull/165)

luc-demeyer committed:
add intrastat module

danielduqma committed:
[11.0][ADD] l10n_es_intrastat_product_declaration

Commit:
[MIG] l10n_es_intrastat_product: Migration to 12.0

pedrobaeza committed:
[MIG+IMP+FIX] l10n_es_intrastat_report: Finish migration to 12 + several imps/fixes

* Remove unneeded dependency to stock_picking_invoice_link.
* Updated HS Codes source file.
* Optimized compute/import methods.
* Remove useless/obsolete code, data and fields.
* Added Spanish translation.

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Allow invoices with no linked purchases

Previous code was forcing the existence of linked purchase lines which may be not
true for all. As we already have a fallback in declaration code, let's not force
this at location level.

joao-p-marques committed:
[IMP] l10n_es_intrastat_report: black, isort, prettier

joao-p-marques committed:
[MIG] l10n_es_intrastat_report: Migration to 13.0

TT27865

joao-p-marques committed:
[FIX] l10n_es_intrastat_report: Update multi-company rules

Missed step from the migration to v13
Update multi-company security domains and provide migration scripts

TT27865

joao-p-marques committed:
[MIG] l10n_es_intrastat_report: Migration to 14.0

danimv5 committed:
Added translation using Weblate (Catalan)

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Fix tests

Due to upstream change, now invoice date is required.

dosipchuk committed:
[FIX] l10n_es_intrastat_report: additional check if company's country is set

Co-authored-by: Pedro M. Baeza <pedro.baeza@tecnativa.com>

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.1.0.1

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (114 of 114 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_intrastat_report
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_intrastat_report/ca/

victoralmau committed:
[IMP] l10n_es_intrastat_report: Update intrastat transaction data according to documentation.
TT30682

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.1.1.0

ValentinVinagre committed:
[IMP] l10n_es_intrastat_report: adaptacion para la nueva presentacion de 2022

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

victoralmau committed:
[IMP] l10n_es_intrastat_report: Add to notes when missing partner_vat and show error when missing product_origin_country_id in line.
TT29332

victoralmau committed:
[IMP] l10n_es_intrastat_report: Add to notes when missing partner_vat and show error when missing product_origin_country_id in line.
TT29332

victoralmau committed:
[FIX] l10n_es_intrastat_report: Change the link to the codes.
TT34295

victoralmau committed:
[FIX] l10n_es_intrastat_report: Change the link to the codes.
TT34295

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.2.0.1

HaraldPanten commented:
@flachica Si puedes ve haciendo revisiones en las dependencias. Que eso siempre acelera el proceso 👍

flachica commented:
> @flachica Si puedes ve haciendo revisiones en las dependencias. Que eso siempre acelera el proceso 👍

Si, está claro. Es el próximo paso. Tengo que dar con la tecla pues cuando reviso me salen demasiados cambios y no consigo tener claro lo que hizo exactamente quien migró cada dependencia

flachica commented:
Revisiones hechas. Para ver las modificaciones que se realizan respecto al código original es sencillo. Vamos del PR a la rama que lo genera, luego en la parte superior derecha buscamos la palabra Contribute y hacemos click en Open Pull Request. Esto nos ofrece la comparativa que estaba buscando.

flachica commented:
> Hey @flachica, thank you for your Pull Request.
> 
> It looks like some users haven't signed our **C**ontributor **L**icense **A**greement, yet. You can read and sign our full Contributor License Agreement here: http://odoo-community.org/page/cla Here is a list of the users:
> 
> * Fernando [fernando.lachica@greenice.com](mailto:fernando.lachica@greenice.com) (no github login found)
> * oca-ci [oca-ci@odoo-community.org](mailto:oca-ci@odoo-community.org) (no github login found)
> * Manuel Calero [manuel.calero@tecnativa.com](mailto:manuel.calero@tecnativa.com) (no github login found)
> 
> Appreciation of efforts, OCA CLAbot

He modificado el [commit](https://github.com/OCA/l10n-spain/pull/2136/commits/de6bd85597d34479236899fcc7e034d2af882f88) afectado por mi error

victoralmau commented:
Gracias por la aportación.
Será necesario incluir los cambios relativos en https://github.com/OCA/l10n-spain/pull/2133 y https://github.com/OCA/l10n-spain/pull/2148 en el histórico de commits antes de los propios de migración.

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.2.1.0

pedrobaeza commented on l10n_es_intrastat_report/models/l10n_es_intrastat_product_declaration.py:
```diff
@@ -79,6 +79,7 @@ def _update_computation_line_vals(self, inv_line, line_vals, notedict):
                     _("Missing partner vat on invoice %s. ") % (inv_line.move_id.name)
                 ]
                 self._format_line_note(inv_line, notedict, line_notes)
+	return result
```
Indentación incorrecta

flachica replied:
Ya lo corregí. Ahora falta que se vayan incorporando las dependencias

pedrobaeza committed:
[IMP] l10n_es_intrastat_report: 2022 NC codes

Updated from https://sede.agenciatributaria.gob.es/Sede/estadisticas/estadisticas-comercio-exterior/nomenclatura-combinada-ano.html

TT35871

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Test at post-install

Since 81aac30dd2278e43a43cf8fd6cef31e1e8c60f3f, all tests depending on
invoicing base test should be post-install.

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Test at post-install

Since 81aac30dd2278e43a43cf8fd6cef31e1e8c60f3f, all tests depending on
invoicing base test should be post-install.

flachica committed:
[IMP] l10n_es_intrastat_report: black, isort

flachica committed:
[MIG] l10n_es_intrastat_report 15.0

victoralmau commented:
Ahora que ya están todas las dependencias fusionadas, puedes hacer rebase y así poder revisarlo?

Faltará por añadir cambios de 14 en el historial de commits: https://github.com/OCA/l10n-spain/commit/cb6b2b67273f9ba907a8c189be7778cb1d599c19 + https://github.com/OCA/l10n-spain/pull/2403

flachica commented:
Falla algo de ticket bai. Mañana le echaré un ojo con más atención

flachica commented:
Me está costando resolver los problemas con los tests. Contexto:

- El módulo `l10n_es_intrastat_report` necesita de un plan contable genérico. Le he [incluido](https://github.com/flachica/l10n-spain/blob/15.0-mig-l10n_es_intrastat_report/l10n_es_intrastat_report/__manifest__.py#L17) la dependencia a `l10n_generic_coa`
- Al ejecutar los tests me dice que le ponga el tag [post_install](https://github.com/OCA/l10n-spain/actions/runs/3271542685/jobs/5381458228#step:8:872). 
- Cuando hago ese [cambio](https://github.com/flachica/l10n-spain/blob/15.0-mig-l10n_es_intrastat_report/l10n_es_intrastat_report/tests/test_l10n_es_intrastat_report.py#L14) en los tests no los pasa. Supongo que es porque [fallan](https://github.com/OCA/l10n-spain/actions/runs/3271650307/jobs/5381689522#step:8:936) los de Ticketbai previamente pues no hay rastro de que se ejecute ningún test de la clase `TestL10nIntraStatReport`. Busco en el log esa cadena y no la encuentro.

Así que necesito un poquito de luz

pedrobaeza commented:
No debes poner esa dependencia. Pon lo del post_install, eso sí.

flachica commented:
@victoralmau Ya pasan los tests

pedrobaeza commented:
/ocabot migration l10n_es_intrastat_report

pedrobaeza commented:
¿Podrías traerte también f88d7f98a419f476fea855ee35d0ae64d9924de5 ?

victoralmau reviewed (commented):
Ups, creo que el commit https://github.com/OCA/l10n-spain/commit/f88d7f98a419f476fea855ee35d0ae64d9924de5 no era necesario añadirlo aquí porque hace referencia a otro módulo diferente.

Por lo demás, en un principio estaría todo correcto.

pedrobaeza commented:
Perdón, error garrafal mío...

flachica commented:
He rebasado la rama y los tests fallarán por algo de los .js (supongo por no haberlos renombrado a esm.js) del módulo [pos de ticketbai](https://github.com/OCA/l10n-spain/tree/15.0/l10n_es_ticketbai_pos)

flachica commented:
Disculpad, el pre-commit ha sabido resolverlo. En local me estaba dando problemas pero parece que runboat lo interpreta correctamente

pedrobaeza commented on l10n_es_intrastat_report/__manifest__.py:
```diff
@@ -24,6 +24,7 @@
         "views/l10n_es_intrastat_product.xml",
         "wizards/l10n_es_intrastat_code_import.xml",
     ],
+    "external_dependencies": {"python": ["xlrd"]},
```
You don't need to add this one, as it's an Odoo base library:

https://github.com/odoo/odoo/blob/3cbb5ad003aa58ed85df182b136ed38d9a0e4ede/requirements.txt#L46-L47

flachica replied:
Vaya, de repente pre-commit empezó a darme mensajes que no se reflejaban en runboat. Debe estar relacionado con los cambios que hice en la rama 16.0 de [oca/connector-prestashop](https://github.com/oca/connector-prestashop/)

pedrobaeza commented on l10n_es_intrastat_report/readme/CONTRIBUTORS.rst:
```diff
@@ -12,3 +12,7 @@
+
+* `GreenIce <https://www.greenice.com>`__:
+
+  * Fernando La Chica <fernandolachica@gmail.com>
```
If you put the company, it seems inconsistent to put a personal email inside. And I recommend you to not put for avoiding "spam"

pedrobaeza commented:
/ocabot merge nobump
